js/lib/widget.js

```diff
@@ -34,15 +34,16 @@ var RemoteFrameBufferModel = widgets.DOMWidgetModel.extend({
         // For the widget
         css_width: '500px',
         css_height: '300px',
-        resizable: true
+        resizable: true,
+        has_visible_views: false
     }),
 
     initialize: function () {
         RemoteFrameBufferModel.__super__.initialize.apply(this, arguments);
         window.rfb_model = this; // Debug
-        // Keep a list if img elements, and auto-refresh it.
+        // Keep a list if img elements. Gets updated when a widget is created, and every 5s.
         this.img_elements = [];
-        window.setInterval(this.resolve_img_elements.bind(this), 1000);
+        window.setInterval(this.collect_view_img_elements.bind(this), 5000);
```

Comment:
Why the large increase in the interval?

Reply:
This interval is a failsafe, now more so than before this PR: In theory the `collect_view_img_elements` is called when needed: e.g. when a new view is created. This PR also calls it when a view is destroyed, so it should generally Just Work. But I'm not confident that the call works with every possible way that a view can be removed.

Reply:
Will add a comment.